arches/app/utils/frontend_configuration_utils.py

```diff
@@ -0,0 +1,198 @@
+import json
+import os
+import re
+import site
+import sys
+
+from django.conf import settings
+from django.urls import get_resolver, URLPattern, URLResolver
+from django.urls.resolvers import RegexPattern, RoutePattern, LocalePrefixPattern
+
+from arches.settings_utils import list_arches_app_names, list_arches_app_paths
+
+
+def generate_frontend_configuration():
+    try:
+        _generate_frontend_configuration_directory()
+        _generate_urls_json()
+        _generate_webpack_configuration()
+        _generate_tsconfig_paths()
+    except Exception as e:
+        # Ensures error message is shown if error encountered
+        sys.stderr.write(str(e))
+        raise e
+
+
+def _generate_frontend_configuration_directory():
+    destination_dir = os.path.realpath(
+        os.path.join(_get_base_path(), "..", "frontend_configuration")
+    )
+
+    os.makedirs(destination_dir, exist_ok=True)
+
+
+def _generate_urls_json():
+    def generate_human_readable_urls(patterns, prefix="", namespace="", result={}):
+        def join_paths(*args):
+            return "/".join(filter(None, (arg.strip("/") for arg in args)))
```

Comment:
We seem to be losing the trailing slash on some routes. Leaving aside whether it's a good idea to have trailing slashes required in your routes, if projects _do_ have that, I think we should leave it to them (and whatever third-party packages even outside of arches they depend on).

For instance, with status quo, we have on javascript.htm:
```
reorder_cards="/reorder_cards/"
```
but now in urls.json:
```
    "reorder_cards": "/reorder_cards",
```
giving:
```js
> _arches_utils_generate_arches_url_ts__WEBPACK_IMPORTED_MODULE_11__["default"]('reorder_cards')
'/reorder_cards'
```

Which will cause a redirect (or not) depending on the value of `APPEND_SLASH` and yada yada.

Can we avoid a behavior change here?

Reply:
nice catch, took a little bit of thinking but made that adjustment 👍